src/zope/schema/_bootstrapinterfaces.py

```diff
@@ -25,17 +27,17 @@ class StopValidation(Exception):
     a way for the validator to save time.
     """
 
-
+@total_ordering
 class ValidationError(zope.interface.Invalid):
     """Raised if the Validation process fails."""
 
     def doc(self):
         return self.__class__.__doc__
 
-    def __cmp__(self, other):
+    def __lt__(self, other):
         if not hasattr(other, 'args'):
             return -1
```

Comment:
I'm not sure what's the right thing to do.  Returning `NotImplemented` would lead to a TypeError, which would lead to test failures.  Should the test be changed to expect a TypeError when comparing an exception to something else, including on Python 2?  Maybe?  I don't know.

Reply:
Yeah, I don't know how that got left at -1. Thanks. I'll poke at it some more. My intent was to keep the behaviour as close to Python 2 as possible to help avoid surprises, but maybe that's not possible.

Reply:
Well, so returning True there keeps the tests passing (which makes sense---the return is auto-converted to a bool, and -1 is True). It vaguely makes sense that this object wants to sort itself less than anything else, and it's consistent with its `__eq__` behaviour. What do you think @mgedmin? Good enough?

Reply:
`True` is good enough for me.

(I'd also be tempted to add a comment mentioning that there's no particular reason or justification, other than the usual non-specific fear of backwards-incompatible changes maybe breaking something for someone somewhere.)